sysbuild/CMakeLists.txt

```diff
@@ -41,6 +41,114 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
     set_config_string(${DEFAULT_IMAGE} CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS
                       "--pad-header --header-size ${SB_CONFIG_PM_MCUBOOT_PAD}"
     )
+
+    if(NOT DEFINED mcuboot_BOARD AND "${BOARD}" STREQUAL "nrf9160dk_nrf9160_ns")
+      # MCUboot must run in secure mode on the nRF9160
+      if(DEFINED BOARD_REVISION)
+        set_target_properties(mcuboot PROPERTIES BOARD nrf9160dk_nrf9160@${BOARD_REVISION})
+      else()
+        set_target_properties(mcuboot PROPERTIES BOARD nrf9160dk_nrf9160)
+      endif()
+    elseif(NOT DEFINED mcuboot_BOARD AND "${BOARD}" STREQUAL "thingy91_nrf9160_ns")
+      # MCUboot must run in secure mode on the nRF9160
+      if(DEFINED BOARD_REVISION)
+        set_target_properties(mcuboot PROPERTIES BOARD thingy91_nrf9160@${BOARD_REVISION})
+      else()
+        set_target_properties(mcuboot PROPERTIES BOARD thingy91_nrf9160)
```

Comment:
I know this is temporary, but there has to be a more efficient way to tabulate this.

Reply:
It needs to check board revision for each, so each board will have 2 flows, I don't see a way to reduce that

Reply:
I was thinking of condensing it like this for a start:
```cmake
if(BOARD STREQUAL "nrf9160dk_nrf9160_ns")
  set(board_secure nrf9160dk_nrf9160)
elseif(BOARD STREQUAL "thingy91_nrf9160_ns")
  set(board_secure thingy91_nrf9160)
elseif(BOARD STREQUAL "nrf5340dk_nrf5340_cpuapp_ns")
  set(board_secure nrf5340dk_nrf5340_cpuapp)
elseif(BOARD STREQUAL "thingy53_nrf5340_cpuapp_ns")
  set(board_secure thingy53_nrf5340_cpuapp)
endif()

if(DEFINED board_secure AND DEFINED BOARD_REVISION)
  set(board_secure ${board_secure}@${BOARD_REVISION})
endif()

...

if(NOT DEFINED mcuboot_BOARD AND DEFINED board_secure)
  # MCUboot must run in secure mode
  set_target_properties(mcuboot PROPERTIES BOARD ${board_secure})
endif()

...

if(NOT DEFINED b0_BOARD AND DEFINED board_secure)
  # b0 must run in secure mode
  set_target_properties(b0 PROPERTIES BOARD ${board_secure})

  if(SB_CONFIG_BOOTLOADER_MCUBOOT)
    set_target_properties(s1_image PROPERTIES BOARD ${board_secure})
  endif()
endif()
```

Reply:
Updated, thanks